python/ray/util/sgd/v2/trainer.py

```diff
@@ -124,7 +127,22 @@ def run(self,
         train_func = self._get_train_func(train_func, config)
         # TODO(matt): Set default callbacks.
         callbacks = [] if callbacks is None else callbacks
-        return self._executor.run(train_func)
+
+        try:
+            self._executor.start_training(train_func)
+            while True:
+                intermediate_results = self._executor.fetch_next_result()
+                if intermediate_results is None:
+                    break
+                else:
+                    for callback in callbacks:
+                        callback.handle_result(intermediate_results)
+            return self._executor.finish_training()
+        except InactiveWorkerGroupError:
+            raise RuntimeError(
+                "This Trainer is not active. It is either shutdown already or "
+                "never started in the first place. Either create a new "
+                "Trainer or start this one.")
```

Comment:
raise ... from exc?

Reply:
Ah yeah good point. I added a raise from None here. I don't think we want to expose `InactiveWorkerGroupError` to the user.